app/backend/db/crud/date_datas.py

```diff
@@ -44,26 +44,36 @@ def get_multiple_users_dates_data(
 
 
 def set_new_date_data(
-    db: Session, 
-    body=schemas.PostDateDataBody
+    db: Session,
+    user_id: int,
+    start_date: date,
+    end_date: date = None,
+    state: schemas.AnswerStateTypes = None,
+    reason: str = None,
+    reported_by_id: int = None,
+    reported_time: datetime = None
 ) -> schemas.RangeDatesResponse:
 
     dates_data = []
-    for date in daterange(body.start_date, body.end_date):
+    for date in daterange(start_date, end_date):
         if reason is not None:
-            reason = get_reason_by_name(db, body.reason)
+            reason = get_reason_by_name(db, reason)
+        
+        elif state.name == 'not_here':
+            raise RuntimeError('Cannot sign as "not_here"' 
+                               'with no reason.')
 
         dates_data.append(DateData(
                 date=date, 
-                user_id=body.user_id,
-                state=body.state, 
+                user_id=user_id,
+                state=state, 
                 reason=reason, 
-                reported_by=body.reported_by, 
-                reported_time=body.reported_time
+                reported_by=reported_by, 
+                reported_time=reported_time
             ))
     db.add_all(dates_data)
```

Comment:
ok you got one thing wrong is that u forgot to actually create DateDetails instance if not exists

Reply:
i thought it makes one automatically :P 
done.